src/main/java/com/bugsnag/android/Client.java

```diff
@@ -61,7 +62,10 @@ public Client(Context androidContext, String apiKey, boolean enableExceptionHand
         // Attempt to load API key from AndroidManifest.xml
         try {
             ApplicationInfo ai = appContext.getPackageManager().getApplicationInfo(appContext.getPackageName(), PackageManager.GET_META_DATA);
-            apiKey = ai.metaData.getString("com.bugsnag.android.API_KEY");
+            String manifestApiKey = ai.metaData.getString("com.bugsnag.android.API_KEY");
+            if(!TextUtils.isEmpty(manifestApiKey)) {
```

Comment:
Interestingly @JakeWharton I cannot reproduce your issue, in my tests I always fall into the `catch` block when trying to load the api key from the manifest.

Also, I think your suggested approach is better.